src/test/java/uk/gov/pay/directdebit/payments/services/GoCardlessServiceTest.java

```diff
@@ -85,9 +102,16 @@
             .toEntity();
 
 
-    GoCardlessPayment goCardlessPayment = aGoCardlessPaymentFixture().withTransactionId(payment.getId()).toEntity();
     BankAccountDetails bankAccountDetails = new BankAccountDetails(ACCOUNT_NUMBER, SORT_CODE);
 
+    @Before
+    public void setUp() {
+        service = new GoCardlessService(mockedGoCardlessClientFactory, mockedGoCardlessCustomerDao, mockedPaymentDao);
+        when(mockedGoCardlessClientFactory.getClientFor(Optional.of(gatewayAccountFixture.getAccessToken()))).thenReturn(mockedGoCardlessClientFacade);
```

Comment:
passing in an Optional parameter is discouraged

Reply:
Agreed, it is existing and we'll address it if we have time in the next PR(s)....